pages/instructorFileEditor/instructorFileEditor.ejs

```diff
@@ -1,78 +1,155 @@
-                        <button type="button" class="close" data-dismiss="alert" aria-label="Close">
-                            <span aria-hidden="true">&times;</span>
-                        </button>
+                </div>
+                <div class="card-body p-0 row">
+                    <div class="container">
+                        <% if (fileEdit.alertResults) { %>
+                        <div class="alert <% if (fileEdit.didSave && fileEdit.didSync) { %>alert-success<% } else { %>alert-danger<% } %> alert-dismissible fade show m-2" role="alert">
+                            <div class="row align-items-center">
+                                <div class="col-auto">
+                                    <% if (fileEdit.didSave) { %>
+                                        <% if (fileEdit.didSync) { %>
+                                            File was both saved and synced successfully.
+                                        <% } else { %>
+                                            File was saved, but failed to sync.
+                                        <% } %>
+                                    <% } else { %>
+                                        Failed to save and sync file.
+                                    <% } %>
+                                </div>
+                                <% if (fileEdit.jobSequenceId != null) { %>
+                                <div class="col-auto">
+                                    <a class="btn btn-secondary btn-sm" data-toggle="collapse" href="#results-<%= fileEdit.uuid %>" role="button">Detail</a>
+                                </div>
+                                <% } %>
+                                <% if (fileEdit.failedPush) { %>
+                                <div class="col-auto">
+                                    <a class="btn btn-primary btn-sm" href="<%= urlPrefix %>/syncs" role="button" target="_blank" rel="noreferrer">Open Sync Page in New Window</a>
+                                </div>
+                                <% } %>
+                                <button type="button" class="close" data-dismiss="alert" aria-label="Close">
+                                    <span aria-hidden="true">&times;</span>
+                                </button>
+                            </div>
+                            <% if (fileEdit.failedPush) { %>
+                            <div class="row mt-4">
+                                <div class="col-auto">
+                                    In particular, it looks like there was a failure to push your changes to the remote git repo. The most likely cause is that another user made changes to other course files while you were editing. In this case, you should open the sync page in a new browser window, pull from remote git repo, close the window, and continue editing.
```

Comment:
Could we add a button to directly kick off a sync in a new tab? So we could say "Click here to sync from GitHub in a new tab, then return here and continue editing." or the like.

Reply:
I definitely agree that what users will want here is a button that means “just fix it”. Ideally we would have a button that POSTs back to the current page to trigger the sequence of pull/sync/save/sync.

However, let’s leave that for another branch (hopefully soon!). Could we just make the text “sync page” in the paragraph above be a link to the sync page? I know that triggering a sync via that would be a bit ugly, so we could also skip that for now.

Reply:
One-click recovery = implemented.